datamodels/note_book.py

```diff
@@ -186,8 +188,10 @@ def find_hobby(self, hobby_: str) -> list[Note]:
         """
         notes = list(
             filter(
-                lambda note_: any(_hobby.value == hobby_ for _hobby in note_.hobbies),
-                self.data.values()
+                lambda note_: any(
+                    _hobby.value.lower() == hobby_.lower() for _hobby in note_.hobbies
```

Comment:
Use casefold() instead of lower() before comparison. This will cover some encoding shenanigans

Reply:
got it for casefold() instead of lower(), but wdym with "before comparison"

Reply:
When comparing values and want it to be done case insensitively use
```
value.casefold() == other.casefold()
```
Instead of
```
value.lower() == other.value()
```

This is needed because casefold() enforces a certain encoding mapping whilst also converting characters from upper to lower case.

This will ensure that two values will be found equal even if some character in one of them was encoded a bit differently in Unicode.